ambry-messageformat/src/main/java/com.github.ambry.messageformat/UndeleteMessageFormatInputStream.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.messageformat;
+
+import com.github.ambry.store.StoreKey;
+import java.nio.ByteBuffer;
+
+
+/**
+ * Represents a message that consists of the undelete record.
+ * This format is used to delete a blob
+ *
+ *  - - - - - - - - - - - - -
+ * |     Message Header      |
+ *  - - - - - - - - - - - - -
+ * |       blob key          |
+ *  - - - - - - - - - - - - -
+ * |      Update Record      |
+ *  - - - - - - - - - - - - -
+ *
+ */
+public class UndeleteMessageFormatInputStream extends MessageFormatInputStream {
+  public UndeleteMessageFormatInputStream(StoreKey key, short accountId, short containerId, long updateTimeInMs)
+      throws MessageFormatException {
+    int headerSize = MessageFormatRecord.getHeaderSizeForVersion(MessageFormatRecord.headerVersionToUse);
+    int recordSize = MessageFormatRecord.Update_Format_V3.getRecordSize(SubRecord.Type.UNDELETE);
+    buffer = ByteBuffer.allocate(headerSize + key.sizeInBytes() + recordSize);
+    MessageFormatRecord.MessageHeader_Format_V2.serializeHeader(buffer, recordSize,
```

Comment:
Looks like you decide to go with `MessageHeader_Format_V2`, which means `lifeVersion` is not tracked in message header. Is this intended?